source/agora/common/EnrollmentManager.d

```diff
@@ -0,0 +1,353 @@
+/*******************************************************************************
+
+    Contains supporting code for enrollment process.
+
+    Copyright:
+        Copyright (c) 2019 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.common.EnrollmentManager;
+
+import agora.common.crypto.ECC;
+import agora.common.crypto.Key;
+import agora.common.crypto.Schnorr;
+import agora.common.crypto.Schnorr : Signature;
+import agora.common.Deserializer;
+import agora.common.Hash;
+import agora.common.Serializer;
+import agora.consensus.data.Enrollment;
+import agora.utils.Log;
+
+import d2sqlite3.database;
+import d2sqlite3.library;
+import d2sqlite3.results;
+import d2sqlite3.sqlite3;
+
+import std.algorithm;
+import std.file;
+import std.path;
+import std.stdio;
+
+mixin AddLogger!();
+
+/*******************************************************************************
+
+    Handle enrollment data and manage the validators set
+
+*******************************************************************************/
+
+public class EnrollmentManager
+{
+    /// SQLite db instance
+    private Database db;
+
+    /// Node's key pair
+    private KeyPair node_key_pair;
+
+    /// Random seed
+    private Scalar random_seed_src;
+
+    /// Random key for enrollment
+    private Pair signature_noise;
+
+    /// Enrollment data object
+    private Enrollment data;
+
+    /***************************************************************************
+
+        Constructor
+
+        Params:
+            db_path = path to the database file, or in-memory storage if
+                        :memory: was passed
+            key_pair = the keypair of the owner node
+
+    ***************************************************************************/
+
+    public this (string db_path, KeyPair key_pair)
+    {
+        this.db = Database(db_path);
+
+        // create the table for validator set if it doesn't exist yet
+        this.db.execute("CREATE TABLE IF NOT EXISTS validator_set " ~
+            "(key BLOB PRIMARY KEY, val BLOB NOT NULL)");
+
+        // create the table for enrollment data for a node itself
+        this.db.execute("CREATE TABLE IF NOT EXISTS node_enroll_data " ~
+            "(key CHAR(128) PRIMARY KEY, val BLOB NOT NULL)");
+
+        // set node's KeyPair object
+        this.node_key_pair = key_pair;
+
+        // load signature noise
+        auto results = this.db.execute("SELECT val FROM node_enroll_data " ~
+            "WHERE key = ?", "signature_noise");
+
+        foreach (row; results)
+        {
+            signature_noise = deserializeFull!Pair(row.peek!(ubyte[])(1));
```

Comment:
In the unittest code, all the data saved in the database cannot be shared because the database is in momory as far as I know. So this branch cannot be tested in the unittest code. Would you please show me a clue for that?